src/scripts/tenantDataMigrations/migrate-existing-tenant-data.js

```diff
@@ -0,0 +1,1252 @@
+					const orgCode = String(row.organization_code || '').trim()
+
+					if (orgId && tenant && orgCode) {
+						this.orgLookupCache.set(orgId, {
+							organization_code: orgCode,
+							tenant_code: tenant,
+						})
+					}
+				})
+				.on('end', () => {
+					console.log(`✅ Loaded ${this.orgLookupCache.size} organization mappings`)
+					resolve()
+				})
+				.on('error', reject)
+		})
+	}
+
+	async checkForExistingData() {
+		console.log('🔍 Checking for existing data that needs migration...')
+
+		try {
+			const testQueries = [
+				"SELECT COUNT(*) as count FROM sessions WHERE tenant_code IS NULL OR tenant_code = ''",
+				"SELECT COUNT(*) as count FROM user_extensions WHERE tenant_code IS NULL OR tenant_code = ''",
+				"SELECT COUNT(*) as count FROM organization_extension WHERE tenant_code IS NULL OR tenant_code = ''",
+			]
+
+			let totalRecordsNeedingMigration = 0
+
+			for (const query of testQueries) {
+				try {
+					const results = await this.sequelize.query(query, { type: Sequelize.QueryTypes.SELECT })
+					totalRecordsNeedingMigration += parseInt(results[0].count)
+				} catch (error) {
+					console.log(`⚠️  Could not check table: ${error.message}`)
+				}
+			}
+
+			console.log(`📊 Found ${totalRecordsNeedingMigration} records with NULL/empty tenant_code`)
+
+			return totalRecordsNeedingMigration > 0
+		} catch (error) {
+			console.log(`⚠️  Error checking for existing data: ${error.message}`)
+			return true
+		}
+	}
+
+	async dropForeignKeyConstraints(transaction) {
+		console.log('🔗 Dropping foreign key constraints for data migration...')
+
+		try {
+			const foreignKeys = await this.sequelize.query(
+				`SELECT DISTINCT
+					tc.table_name,
+					tc.constraint_name,
+					string_agg(DISTINCT kcu.column_name, ', ' ORDER BY kcu.column_name) as columns,
+					ccu.table_name AS foreign_table_name,
+					string_agg(DISTINCT ccu.column_name, ', ' ORDER BY ccu.column_name) as foreign_columns
+				FROM information_schema.table_constraints AS tc 
+				JOIN information_schema.key_column_usage AS kcu
+					ON tc.constraint_name = kcu.constraint_name AND tc.table_schema = kcu.table_schema
+				JOIN information_schema.constraint_column_usage AS ccu
+					ON ccu.constraint_name = tc.constraint_name AND ccu.table_schema = tc.table_schema
+				WHERE tc.constraint_type = 'FOREIGN KEY' 
+				AND tc.table_schema = 'public'
+				GROUP BY tc.constraint_name, tc.table_name, ccu.table_name
+				ORDER BY tc.table_name`,
+				{ type: Sequelize.QueryTypes.SELECT, transaction }
+			)
+
+			const droppedConstraints = []
+
+			for (const fk of foreignKeys) {
+				try {
+					await this.sequelize.query(
+						`ALTER TABLE "${fk.table_name}" DROP CONSTRAINT IF EXISTS "${fk.constraint_name}"`,
+						{ transaction }
+					)
+					droppedConstraints.push(fk)
+					console.log(`✅ Dropped FK: ${fk.constraint_name} from ${fk.table_name}`)
+				} catch (error) {
+					console.log(`⚠️  Could not drop FK ${fk.constraint_name}: ${error.message}`)
```

Comment:
Why is this best effort? If we are unable to drop FKs, the script will break, right?